scripts/exregional_run_prepstart.sh

```diff
@@ -0,0 +1,278 @@
+#
+# Specify the set of valid argument names for this script/function.  
+# Then process the arguments provided to this script/function (which 
+# should consist of a set of name-value pairs of the form arg1="value1",
+# etc).
+#
+#-----------------------------------------------------------------------
+#
+valid_args=( "CYCLE_DIR" "ANALWORKDIR" "FG_ROOT")
+process_args valid_args "$@"
+#
+#-----------------------------------------------------------------------
+#
+# For debugging purposes, print out values of arguments passed to this
+# script.  Note that these will be printed out only if VERBOSE is set to
+# TRUE.
+#
+#-----------------------------------------------------------------------
+#
+print_input_args valid_args
+#
+#-----------------------------------------------------------------------
+#
+# Load modules.
+#
+#-----------------------------------------------------------------------
+#
+case $MACHINE in
+#
+"WCOSS_C" | "WCOSS")
+#
+  ;;
+#
+"THEIA")
+#
+  ;;
+#
+"HERA")
+  ;;
+#
+"JET")
+  NCKS=ncks
+  ;;
+#
+"ODIN")
+#
+  ;;
+#
+esac
+#
+#-----------------------------------------------------------------------
+#
+# Extract from CDATE the starting year, month, day, and hour of the
+# forecast.  These are needed below for various operations.
+#
+#-----------------------------------------------------------------------
+#
+START_DATE=`echo "${CDATE}" | sed 's/\([[:digit:]]\{2\}\)$/ \1/'`
+
+YYYYMMDDHH=`date +%Y%m%d%H -d "${START_DATE}"`
+JJJ=`date +%j -d "${START_DATE}"`
+
+YYYY=${YYYYMMDDHH:0:4}
+MM=${YYYYMMDDHH:4:2}
+DD=${YYYYMMDDHH:6:2}
+HH=${YYYYMMDDHH:8:2}
+YYYYMMDD=${YYYYMMDDHH:0:8}
+#
+#-----------------------------------------------------------------------
+#
+# go to INPUT directory.
+# prepare initial conditions for 
+#     cold start if BKTYPE=1 
+#     warm start if BKTYPE=0
+#       the previous 6 cycles are searched to find the restart files
+#       valid at this time from the closet previous cycle.
+#
+#-----------------------------------------------------------------------
+
+cd_vrfy ${ANALWORKDIR}
+
+if [ ${BKTYPE} -eq 1 ]; then  # cold start, use prepare cold strat initial files from ics
+  bkpath=${CYCLE_DIR}/ics
+  if [ -r "${bkpath}/gfs_data.tile7.halo0.nc" ]; then
+    cp_vrfy ${bkpath}/gfs_bndy.tile7.000.nc gfs_bndy.tile7.000.nc        
+    cp_vrfy ${bkpath}/gfs_ctrl.nc gfs_ctrl.nc        
+    cp_vrfy ${bkpath}/gfs_data.tile7.halo0.nc gfs_data.tile7.halo0.nc        
+    cp_vrfy ${bkpath}/sfc_data.tile7.halo0.nc sfc_data.tile7.halo0.nc        
+    print_info_msg "$VERBOSE" "cold start from $bkpath"
+  else
+    print_err_msg_exit "Error: cannot find cold start initial condition from : ${bkpath}"
+  fi
+else
+  YYYYMMDDHHmInterv=`date +%Y%m%d%H -d "${START_DATE} ${DA_CYCLE_INTERV} hours ago"`
+  bkpath=${FG_ROOT}/${YYYYMMDDHHmInterv}/fcst_fv3lam/RESTART  # cycling, use background from RESTART
+
+#   let us figure out which backgound is available
+  restart_prefix=${YYYYMMDD}.${HH}0000.
```

Comment:
Consider including quotes on this definition of `restart_prefix` so that we explicitly understand the `.` at the end should be included.

Reply:
I agree. it is better to write as:
restart_prefix=`"`${YYYYMMDD}.${HH}0000.`"`
It is easy to neglect the `.`

Reply:
It's easier to notice quotes and recognize the need for the `.` than to read a comment. I'm as guilty as anyone of ignoring the documentation. :(

Reply:
Added.